services/submission.js

```diff
@@ -610,6 +610,44 @@ class Submission {
         return ValidationHandler.success(`${res.filter(result => result.status === 'fulfilled').length} extra files deleted`);
     }
 
+    /**
+     * deleteInactiveSubmission
+     * description: overnight job to set inactive submission status to "Deleted", delete related data and files
+     */
+    async deleteInactiveSubmissions(){
+        //get target inactive date, current date - config.inactive_submission_days (default 120 days)
+        var target_inactive_date = new Date();
+        target_inactive_date.setDate(target_inactive_date.getDate() - config.inactive_submission_days);
+        const query = [{"$match": {"status": IN_PROGRESS, "accessedAt": {"$exists": true, "$ne": null, "$lte": target_inactive_date}}}];
+        try {
+            const inactive_subs = await this.submissionCollection.aggregate(query);
+            if (!inactive_subs || inactive_subs.length === 0) {
+                console.debug("No inactive submission found.")
+                return "No inactive submissions";
+            }
+            let failed_delete_subs = []
+            //delete related data and files
+            for (const sub of inactive_subs) {
+                try {
+                    const result = await this.s3Service.deleteDirectory(sub.bucketName, sub.rootPath);
+                    if (result === true) {
+                        await this.dataRecordService.deleteMetadataByFilter({"submissionID": sub._id});
```

Comment:
Let's also delete batches of the submission here.

Reply:
Updated by adding delete related batch.